extra_data/write_cxi.py

```diff
@@ -163,7 +163,26 @@ def collect_data(self):
 
         return layouts
 
-    def write(self, filename):
+    @staticmethod
+    def _fill_value(path, layout, fillvalues):
+        array_type = layout[path].dtype.type(0).item()
+        fallback = np.nan if isinstance(array_type, float) else 0
+        value = fillvalues.get(path, fallback)
+        return layout[path].dtype.type(value)
```

Comment:
this is a bit convoluted as `create_virtual_dataset` seems to silently convert `nan` to `0` for integer arrays, but this will raise an error if someone tries to fill an int array with nan instead of silently converting it to zeros.

Reply:
I see your attempt is more professional 

> > How would this dictionary argument look like from the user perspective, at the CLI:
> > `extra-data-make-virtual cxi <RUN-PATH> -o <OUT-NAME> --fill-value=data:0` ?
> 
> There is an example in the PR description #59



> > How would this dictionary argument look like from the user perspective, at the CLI:
> > `extra-data-make-virtual cxi <RUN-PATH> -o <OUT-NAME> --fill-value=data:0` ?
> 
> There is an example in the PR description #59

Very well, thanks, right on top and I missed it ...